src/main/java/ch/njol/skript/util/Timespan.java

```diff
@@ -32,97 +37,134 @@
 
-	private final static Noun m_tick = new Noun("time.tick");
-	private final static Noun m_second = new Noun("time.second");
-	private final static Noun m_minute = new Noun("time.minute");
-	private final static Noun m_hour = new Noun("time.hour");
-	private final static Noun m_day = new Noun("time.day");
-	private final static Noun m_week = new Noun("time.week");
-	private final static Noun m_month = new Noun("time.month");
-	private final static Noun m_year = new Noun("time.year");
-	final static Noun[] names = {m_tick, m_second, m_minute, m_hour, m_day, m_week, m_month, m_year};
-	final static long[] times = {50L, 1000L, 1000L * 60L, 1000L * 60L * 60L, 1000L * 60L * 60L * 24L,  1000L * 60L * 60L * 24L * 7L,  1000L * 60L * 60L * 24L * 30L,  1000L * 60L * 60L * 24L * 365L};
-	final static HashMap<String, Long> parseValues = new HashMap<>();
+
+	public enum TimePeriod {
+
+		TICK("time.tick", 50L),
+		SECOND("time.second", 1000L),
+		MINUTE("time.minute", SECOND.time * 60L),
+		HOUR("time.hour", MINUTE.time * 60L),
+		DAY("time.day", HOUR.time * 24L),
+		WEEK("time.week", DAY.time * 7L),
+		MONTH("time.month", DAY.time * 30L), // Who cares about 28, 29 or 31 days?
+		YEAR("time.year", DAY.time * 365L);
+
+		private final Noun name;
+		private final long time;
+
+		TimePeriod(String name, long time) {
+			this.name = new Noun(name);
+			this.time = time;
+		}
+
+		public long getTime() {
+			return time;
+		}
+
+	}
+
+	private static final List<NonNullPair<Noun, Long>> SIMPLE_VALUES = Arrays.asList(
+		new NonNullPair<>(TimePeriod.YEAR.name, TimePeriod.YEAR.time),
+		new NonNullPair<>(TimePeriod.MONTH.name, TimePeriod.MONTH.time),
+		new NonNullPair<>(TimePeriod.WEEK.name, TimePeriod.WEEK.time),
+		new NonNullPair<>(TimePeriod.DAY.name, TimePeriod.DAY.time),
+		new NonNullPair<>(TimePeriod.HOUR.name, TimePeriod.HOUR.time),
+		new NonNullPair<>(TimePeriod.MINUTE.name, TimePeriod.MINUTE.time),
+		new NonNullPair<>(TimePeriod.SECOND.name, TimePeriod.SECOND.time)
+	);
+
+	private static final Map<String, Long> PARSE_VALUES = new HashMap<>();
+
 	static {
 		Language.addListener(new LanguageChangeListener() {
 			@Override
 			public void onLanguageChange() {
-				for (int i = 0; i < names.length; i++) {
-					parseValues.put(names[i].getSingular().toLowerCase(Locale.ENGLISH), times[i]);
-					parseValues.put(names[i].getPlural().toLowerCase(Locale.ENGLISH), times[i]);
+				for (TimePeriod time : TimePeriod.values()) {
+					PARSE_VALUES.put(time.name.getSingular().toLowerCase(Locale.ENGLISH), time.getTime());
+					PARSE_VALUES.put(time.name.getPlural().toLowerCase(Locale.ENGLISH), time.getTime());
 				}
 			}
 		});
 	}
+
+	public static final Pattern TIMESPAN_PATTERN = Pattern.compile("^(\\d+):(\\d\\d)(:\\d\\d){0,2}(?<ms>\\.\\d{1,4})?$");
+	public static final Pattern TIMESPAN_NUMBER_PATTERN = Pattern.compile("^\\d+(\\.\\d+)?$");
+	public static final Pattern TIMESPAN_SPLIT_PATTERN = Pattern.compile("[:.]");
```

Comment:
```suggestion
	public static final Pattern TIMESPAN_PATTERN = Pattern.compile("^(\\d+):(\\d\\d)(:\\d\\d){0,2}(?<ms>\\.\\d{1,4})?$");
	public static final Pattern TIMESPAN_NUMBER_PATTERN = Pattern.compile("^\\d+(\\.\\d+)?$");
	public static final Pattern TIMESPAN_SPLIT_PATTERN = Pattern.compile("[:.]");

```
also, is there a reason for these to be public?